File: server/src/main/java/pl/bk/pizza/store/application/dto/order/NewOrderDTO.java
package pl.bk.pizza.store.application.dto.order;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;

@AllArgsConstructor
@NoArgsConstructor
@Getter
public class NewOrderDTO
{
    private String email;
    private DeliveryInfoDTO deliveryInfoDTO;
}

File: client/pizza-store/src/app/order/newOrderDTO.ts
import {DeliveryInfoDTO} from "./deliveryInfoDTO";

export class NewOrderDTO {

  email: String;
  deliveryInfoDTO: DeliveryInfoDTO;

  constructor(data:any) {
    this.email = data.email;
    this.deliveryInfoDTO = new DeliveryInfoDTO(data.deliveryInfo);
  }
}
